packages/@react-native-windows/cli/src/runWindows/utils/vstools.ts

```diff
@@ -140,14 +165,34 @@ export function addProjectToSolution(
     'EndProject',
   ];
 
-  if (!linesContainsBlock(slnLines, projectEntryBlock)) {
+  const projectEntryBlockRegex = [
+    new RegExp(
+      `Project\\("${projectTypeGuid}"\\) = "${project.projectName}", "(.*)", "${projectGuid}"`,
+    ),
+    'EndProject',
+  ];
+
+  const blockIndex = linesContainsBlock(slnLines, projectEntryBlockRegex);
+
+  if (blockIndex < 0) {
+    // Regex didn't match, insert
     if (verbose) {
-      console.log(chalk.yellow('Missing project entry block.'));
+      console.log(chalk.yellow('Missing project entry block, inserting.'));
```

Comment:
consider printing out the `projectName` / path for debugging purposes

Reply:
The logging already outputs the project name earlier on.